rastervision/evaluation/semantic_segmentation_evaluation.py

```diff
@@ -12,6 +12,40 @@ def __init__(self, class_map):
         super().__init__()
         self.class_map = class_map
 
+    def compute_vector(self, gt_filename, pred_filename, mode, class_id):
+        import mask_to_polygons.vectorification as vectorification
+        import mask_to_polygons.processing.score as score
+
+        ground_truth = vectorification.geometries_from_geojson(gt_filename)
+        predictions = vectorification.geometries_from_geojson(pred_filename)
+        if len(ground_truth) > 0 and len(predictions) > 0:
+            results = score.spacenet(predictions, ground_truth)
+
+            true_positives = results['tp']
+            false_positives = results['fp']
+            false_negatives = results['fn']
+            precision = float(true_positives) / (
+                true_positives + false_positives)
+            recall = float(true_positives) / (true_positives + false_negatives)
+            if precision + recall != 0:
+                f1 = 2 * (precision * recall) / (precision + recall)
+            else:
+                f1 = 0.0
+            count_error = int(false_positives + false_negatives)
+            gt_count = len(ground_truth)
+            class_name = 'vector-{}'.format(mode)
+
+            evaluation_item = ClassEvaluationItem(precision, recall, f1,
+                                                  count_error, gt_count,
+                                                  -class_id, class_name)
+
+            if hasattr(self, 'class_to_eval_item') and isinstance(
+                    self.class_to_eval_item, dict):
+                self.class_to_eval_item[-class_id] = evaluation_item
+            else:
+                self.class_to_eval_item = {-class_id: evaluation_item}
```

Comment:
Using negative numbers to indicate vector evals could be confusing to a user and seems kind of hacky. Also, the pattern won't generalize if we added a third way of evaluating things. Instead, you should be able to save to a separate eval file pretty easily as follows. You could just override `SemanticSegmentationEvaluator.process()` to call `super().process()` which would do the normal pixel-wise eval, and then do the vector-specific stuff, saving to a file named `eval-vector.json`. This is related to the other comment re: `SemanticSegmentationEvaluator`

Reply:
I propose making a future work card for this change.

Reply:
Ok

Reply:
https://github.com/azavea/raster-vision/issues/642